src/drivers/driver.ts

```diff
@@ -603,7 +607,42 @@ export abstract class CMakeDriver implements vscode.Disposable {
       // Expand all important paths
       await this._refreshExpansions();
 
+      const timeStart: number = new Date().getTime();
       const retc = await this.doConfigure(expanded_flags, consumer);
+      const timeEnd: number = new Date().getTime();
+
+      const cmakeVersion = this.cmake.version;
+      const telemetryProperties: {[key: string]: string} = {
+        CMakeExecutableVersion: cmakeVersion ? `${cmakeVersion.major}.${cmakeVersion.minor}.${cmakeVersion.patch}` : '',
+        CMakeGenerator: this.generatorName || '',
+        ConfigType: this.isMultiConf ? 'MultiConf' : this.currentBuildType || '',
+      };
+      const azureSphereTargetApiSet = this.cmakeCacheEntries.get('AZURE_SPHERE_TARGET_API_SET');
```

Comment:
Hmm... I don't think AzureSphere telemetry belongs in here.  Shouldn't your extension log this instead?

Reply:
This is written before that session id discussion. Since we can use session id to query the cmake telemetries for azure sphere projects, I guess we can remove this.

Reply:
Discussed with Oleg, actually this is still the best way for us since this is the only way we can get build telemetry with TargetApiSet.